pkg/nuget/csproj/parse.go

```diff
@@ -0,0 +1,52 @@
+package csproj
+
+import (
+	"encoding/xml"
+
+	"golang.org/x/xerrors"
+
+	dio "github.com/aquasecurity/go-dep-parser/pkg/io"
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+	"github.com/aquasecurity/go-dep-parser/pkg/utils"
+)
+
+type cfgPackageReference struct {
+	XMLName       xml.Name `xml:"PackageReference"`
+	Version       string   `xml:"Version,attr"`
+	Include       string   `xml:"Include,attr"`
+	PrivateAssets string   `xml:"PrivateAssets"`
+}
+
+type config struct {
+	XMLName  xml.Name              `xml:"Project"`
+	Packages []cfgPackageReference `xml:"ItemGroup>PackageReference"`
+}
+
+type Parser struct{}
+
+func NewParser() types.Parser {
+	return &Parser{}
+}
+
+func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency, error) {
+	var cfgData config
+	if err := xml.NewDecoder(r).Decode(&cfgData); err != nil {
+		return nil, nil, xerrors.Errorf("failed to decode .csproj file: %w", err)
+	}
+
+	libs := make([]types.Library, 0)
+	for _, pkg := range cfgData.Packages {
+		if pkg.Include == "" || pkg.PrivateAssets != "" {
```

Comment:
looks like we need use `pkg.PrivateAssets != "all"`here
if i understand correctly: this and other asset tags work with [dependency parts](https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets).
I think we need skip dependency only if all parts are private.

Also i am not sure about `pkg.ExcludeAssets == "all"`.

Reply:
> looks like we need use pkg.PrivateAssets != "all" here

Yes, that seems indeed correct.

PrivateAssets does not precisely correspond to devDependencies, but in practice `PrivateAssets = all` seems to be used for development dependencies.

I also discovered that PrivateAssets could be an attribute on PackageReference instead of a tag, and that it is case insensitive. I'll change the code to case-insensensitively compare either the tag or the attribute to "all".

Reply:
As for ExcludeAssets, there are two common options:

* `ExcludeAssets=all`, this basically means that the dependency isn't a dependency. Nothing is loaded from it. I am not sure why you would want this, but I think we should exclude it.
* `ExcludeAssets=runtime`, this means that the assemblies are not available on runtime, but only during compile time. It's used for analyzers, code generators, etc. So it seems to indicate a development dependency, and we should exclude it.